### PR TITLE
Generate CRI-O CLI man page during build

### PR DESCRIPTION
**Pull request by saschagrunert**
The information from the crio.8 man page have been transferred to the
CLI, which will now auto-generate the man/markdown documentation.

We now add an indicator that the CLI man pages get autogenerated so we
should not edit the markdown file directly.

### CODE REVIEW

File: cmd/crio/config.go [hunk 1]
```diff
@@ -10,12 +10,14 @@ import (
 )
 
 var configCommand = cli.Command{
-	Name:  "config",
-	Usage: "generate crio configuration files",
+	Name: "config",
+	Usage: `Outputs a commented version of the configuration file that would've been used
```
Comment (TomSweeneyRedHat) [inline cmd/crio/config.go]: I think s/could be/would've/

File: cmd/crio/main.go [hunk 2]
```diff
@@ -80,6 +80,20 @@ func catchShutdown(ctx context.Context, cancel context.CancelFunc, gserver *grpc
...
 }
 
+const usage = `OCI-based implementation of Kubernetes Container Runtime Interface Daemon
+
+crio is meant to provide an integration path between OCI conformant runtimes
+and the kubelet. Specifically, it implements the Kubelet Container Runtime
+Interface (CRI) using OCI conformant runtimes. The scope of crio is tied to the
+scope of the CRI.
+
+1. Support multiple image formats including the existing Docker image format.
```
Comment (TomSweeneyRedHat) [inline cmd/crio/main.go]: Can we change "Docker image format" to "Docker and OCI image formats"?

File: completions/zsh/_crio [hunk 3]
```diff
@@ -1,7 +1,9 @@
 _cli_zsh_autocomplete() {
 
   local -a cmds
-  cmds=('config:generate crio configuration files' 'complete:Output shell completion code' 'completion:Output shell completion code' 'wipe:wipe CRI-O's container and image storage' 'help:Shows a list of commands or help for one command' 'h:Shows a list of commands or help for one command')
+  cmds=('complete:Generate bash, fish or zsh completions.' 'completion:Generate bash, fish or zsh completions.' 'man:Generate the man page documentation.' 'markdown:Generate the markdown documentation.' 'md:Generate the markdown documentation.' 'config:Outputs a commented version of the configuration file that would've been used
```
Comment (TomSweeneyRedHat) [inline completions/zsh/_crio]: same could be/would've change here.

File: docs/crio.8.md [hunk 4]
```diff
@@ -90,196 +100,225 @@ crio is meant to provide an integration path between OCI conformant runtimes and
...
-**--cni-plugin-dir**="": CNI plugin binaries directory (default: "/opt/cni/bin/")
+**--cni-plugin-dir**="": CNI plugin binaries directory (default: "")
 
-**--config, -c**="": path to configuration file
+**--config, -c**="": Path to configuration file (default: /etc/crio/crio.conf)
 
 **--conmon**="": Path to the conmon binary, used for monitoring the OCI runtime. Will be searched for using $PATH if empty. (default: "")
 
-**--conmon-cgroup**="": cgroup used for conmon process (default: "pod")
+**--conmon-cgroup**="": Cgroup to be used for conmon process (default: "system.slice")
```
Comment (TomSweeneyRedHat) [inline docs/crio.8.md]: Double check me on this one, but I think Cgroup is a weird one that's supposed to be lower cased 'cgroup'

File: internal/pkg/criocli/criocli.go [hunk 5]
```diff
@@ -266,131 +273,134 @@ func getCrioFlags(defConf *libconfig.Config, systemContext *types.SystemContext)
...
 			Name:      "conmon",
-			Usage:     fmt.Sprintf("path to the conmon executable (default: %q)", defConf.Conmon),
+			Usage:     fmt.Sprintf("Path to the conmon binary, used for monitoring the OCI runtime. Will be searched for using $PATH if empty. (default: %q)", defConf.Conmon),
 			EnvVar:    "CONTAINER_CONMON",
 			TakesFile: true,
 		},
 		cli.StringFlag{
 			Name:   "conmon-cgroup",
-			Usage:  fmt.Sprintf("cgroup used for conmon process (default: %q)", defConf.ConmonCgroup),
+			Usage:  fmt.Sprintf("Cgroup to be used for conmon process (default: %q)", defConf.ConmonCgroup),
```
Comment (TomSweeneyRedHat) [inline internal/pkg/criocli/criocli.go]: cgroup here too

File: cmd/crio/config.go (hunk 1 above)
Comment (saschagrunert) [reply cmd/crio/config.go]: :+1:

File: cmd/crio/main.go (hunk 2 above)
Comment (saschagrunert) [reply cmd/crio/main.go]: :+1:

File: completions/zsh/_crio (hunk 3 above)
Comment (saschagrunert) [reply completions/zsh/_crio]: :+1:

File: docs/crio.8.md (hunk 4 above)
Comment (saschagrunert) [reply docs/crio.8.md]: :+1:

File: internal/pkg/criocli/criocli.go (hunk 5 above)
Comment (saschagrunert) [reply internal/pkg/criocli/criocli.go]: :+1:

**Comment by TomSweeneyRedHat** [comment]
LGTM and happy green test buttons

**Pull request merged**